add configurable log level to errors plugin (#4718)
miekg/coredns: plugin/errors/README.md

~~~
errors {

> 	consolidate DURATION REGEXP <LEVEL?>

Reviewer: ```suggestion
	consolidate DURATION REGEXP [LEVEL]...
```

Reviewer: Also do we want multiple levels? Or do we assume the same error text will never be in multiple levels?

Author: I think it does not make sense to log the error message on multiple log levels. The error should be logged either as ERROR, WARNING, DEBUG, or INFO

> 	consolidate DURATION REGEXP [LEVEL]...

Reviewer: ```suggestion
	consolidate DURATION REGEXP [LEVEL]
```
as discussed, there will only be one

Author: 👍 right, fixed

}
~~~

Option `consolidate` allows collecting several error messages matching the regular expression **REGEXP** during **DURATION**. After the **DURATION** since receiving the first such message, the consolidated message will be printed to standard output with

> log level, which is configurable by optional option **LEVEL**, e.g. Supported log levels for **LEVEL** option are `warn`,`error`,`info`,`debug`.

Reviewer: ```suggestion
log level, which is configurable by optional option **LEVEL**. Supported options for **LEVEL**  are `warn`,`error`,`info` and `debug`.
```

Author: thanks 👍 changed as you suggested

. {
    forward . 8.8.8.8
    errors {

>         consolidate 5m ".* i/o timeout$" warn

Reviewer 2: ~This addition should be reflected in the description of the example above it.~

Nevermind - already merged.
